src/node.cc

```diff
@@ -2487,46 +2491,100 @@ struct node_module* get_linked_module(const char* name) {
   return mp;
 }
 
-// DLOpen is process.dlopen(module, filename).
+#ifdef __POSIX__
+
+static void* node_dlhandle(void* handle) {
+  return handle;
+}
+
+static void node_dlclose(void* handle) {
+  dlclose(handle);
+}
+
+static void* node_dlopen(const char* filename, int flags, char** dlerrmsg) {
+  // libuv does not support passing flags to dlopen.
+  void* handle = dlopen(filename, flags);
+  if (handle != nullptr)
+    return handle;
+
+  const char* errmsg = dlerror();
+  *dlerrmsg = strdup((errmsg != nullptr) ? errmsg : "no error");
```

Comment:
I'd make this a `std::string*` and then simply do `if (errmsg != nullptr) *dlerrmsg = errmsg;`  The less manual memory allocation, the better.